code/game/gamemodes/vampire/vampire.dm

```diff
@@ -503,6 +603,14 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 		hud_used.vampire_blood_display.maptext_height = 64
 		hud_used.vampire_blood_display.maptext = "<div align='left' valign='top' style='position:relative; top:0px; left:6px'> U:<font color='#33FF33' size='1'>[mind.vampire.bloodusable]</font><br> T:<font color='#FFFF00' size='1'>[mind.vampire.bloodtotal]</font></div>"
 	handle_vampire_cloak()
+	handle_vampire_menace()
+	handle_vampire_smite()
 	if(istype(loc, /turf/space))
 		check_sun()
+	if(istype(loc, /obj/structure/closet/coffin))
```

Comment:
This is snowflakey, consider putting a proc or var on structures indicating their interactions with vampires